docs/source/common/optimizers.rst

```diff
@@ -390,100 +487,65 @@ Lightning will call each optimizer sequentially:
+       if optimizer_idx == 1:
+           if batch_idx % 4 == 0:
+               optimizer.step(closure=optimizer_closure)
 
-        # update discriminator opt every 4 steps
-        if optimizer_idx == 1:
-            if batch_nb % 4 == 0 :
-               optimizer.step(closure=closure)
-
-Here we add a learning-rate warm up
+Here we add a learning rate warm-up.
 
 .. testcode::
 
-    # learning rate warm-up
-    def optimizer_step(self, current_epoch, batch_nb, optimizer, optimizer_idx, closure, on_tpu=False, using_native_amp=False, using_lbfgs=False):
-        # warm up lr
-        if self.trainer.global_step < 500:
-            lr_scale = min(1., float(self.trainer.global_step + 1) / 500.)
-            for pg in optimizer.param_groups:
-                pg['lr'] = lr_scale * self.hparams.learning_rate
+   # learning rate warm-up
+   def optimizer_step(self, epoch, batch_idx, optimizer, optimizer_idx, optimizer_closure, on_tpu=False, using_native_amp=False, using_lbfgs=False):
+       # skip the first 500 steps
+       if self.trainer.global_step < 500:
+           lr_scale = min(1., float(self.trainer.global_step + 1) / 500.)
+           for pg in optimizer.param_groups:
+               pg['lr'] = lr_scale * self.hparams.learning_rate
 
-        # update params
-        optimizer.step(closure=closure)
+       # update params
+       optimizer.step(closure=optimizer_closure)
 
-.. note:: The default ``optimizer_step`` is relying on the internal ``LightningOptimizer`` to properly perform a step. It handles TPUs, AMP, accumulate_grad_batches and much more ...
+.. note:: The default :meth:`~pytorch_lightning.LightningModule.optimizer_step` is relying on the internal :class:`~pytorch_lightning.core.optimizer.LightningOptimizer` to properly perform a step. It handles TPUs, AMP, gradient accumulation and much more ...
 
-.. testcode::
+.. testcode:: python
 
-    # function hook in LightningModule
-    def optimizer_step(self, current_epoch, batch_nb, optimizer, optimizer_idx, closure, on_tpu=False, using_native_amp=False, using_lbfgs=False):
-      optimizer.step(closure=closure)
+   # function hook in LightningModule
+   def optimizer_step(self, epoch, batch_idx, optimizer, optimizer_idx, optimizer_closure, on_tpu=False, using_native_amp=False, using_lbfgs=False):
+       optimizer.step(closure=optimizer_closure)
 
 .. note:: To access your wrapped Optimizer from ``LightningOptimizer``, do as follow.
 
-.. testcode::
-
-    # function hook in LightningModule
-    def optimizer_step(self, current_epoch, batch_nb, optimizer, optimizer_idx, closure, on_tpu=False, using_native_amp=False, using_lbfgs=False):
-
-      # `optimizer is a ``LightningOptimizer`` wrapping the optimizer.
-      # To access it, do as follow:
-      optimizer = optimizer.optimizer
-
-      # run step. However, it won't work on TPU, AMP, etc...
-      optimizer.step(closure=closure)
-
-
-----------
-
-Using the closure functions for optimization
```

Comment:
This section is completely removed because
1. from 1.2.2, this isn't needed even when using LBFGS-like optimizers
2. It doesn't include `zero_grad` in the closure function, so this example is wrong anyway.